.github/workflows/publish.yml

```diff
@@ -1,24 +1,27 @@
 name: Publish packages on NPM 📦
+
 on:
   push:
     branches:
       - main
+
 jobs:
   publish:
+    timeout-minutes: 20
```

Comment:
I added this to all workflows - I had some rogue actions in the past that were running for a couple of hours, it's IMHO better to put some reasonable limit upfront and fail the action sooner than in a couple of hours :P